submissions/bmukha/frogger/app.js

```diff
@@ -0,0 +1,149 @@
+let highScore = localStorage.getItem('highScore') ?? 0;
+let currentScore = 0;
+const highHolder = document.querySelector('.high');
+const currentHolder = document.querySelector('.current');
+const infoHolder = document.querySelector('.info');
+
+const getRandomNumber = (min, max) =>
+  Math.floor(Math.random() * (max - min) + min);
+
+const directions = ['left', 'right'];
+
+const getRandomDirection = () =>
+  directions[Math.floor(Math.random() * directions.length)];
+
+canvas = {
+  width: 505,
+  height: 606,
+};
+
+class Enemy {
+  constructor(startY, mult) {
+    this.width = 101;
+    this.height = 171;
+    this.direction = getRandomDirection();
+    this.sprite = `images/enemy-bug-${this.direction}.png`;
+    this.minX = -this.width;
+    this.maxX = canvas.width + this.width;
+    this.startX = this.direction === 'right' ? this.minX : this.maxX;
+    this.startY = startY;
+    this.currX = this.startX;
+    this.currY = this.startY;
+    this.mult = this.direction === 'right' ? mult : -mult;
+  }
+  update(dt) {
+    if (
+      player.currX >= this.currX &&
+      player.currX <= this.currX + this.width &&
+      player.currY >= this.currY &&
+      player.currY <= this.currY + 85
+    ) {
+      infoHolder.innerText = `You lost`;
+      currentScore = 0;
+      currentHolder.innerText = `Current score: ${currentScore}`;
```

Comment:
Class shouldn't know anything about environment it operates in (whether it web or node engine).